src/nft/helpers.ts

```diff
@@ -49,30 +54,99 @@ export const nftsFromOperations = (ops: Operation[]): NFT[] => {
 };
 
 export const nftsByCollections = (
-  nfts: NFT[] = [],
+  nfts: Array<ProtoNFT | NFT> = [],
   collectionAddress?: string
-): CollectionWithNFT[] => {
-  const filteredNfts = collectionAddress
-    ? nfts.filter((n) => n.collection.contract === collectionAddress)
-    : nfts;
+): Record<string, Array<ProtoNFT | NFT>> | Array<ProtoNFT | NFT> => {
+  return collectionAddress
+    ? nfts?.filter((n) => n.contract === collectionAddress)
+    : nfts.reduce((acc, nft) => {
+        const { contract } = nft;
 
-  const collectionMap = filteredNfts.reduce(
-    (acc: CollectionMap<CollectionWithNFT>, nft: NFT) => {
-      const { collection, ...nftWithoutCollection } = nft;
+        if (!acc[contract]) {
+          acc[contract] = [];
+        }
+        acc[contract].push(nft);
 
-      if (!acc[collection.contract]) {
-        acc[collection.contract] = { ...collection, nfts: [] };
-      }
-      acc[collection.contract].nfts.push(nftWithoutCollection);
-
-      return acc;
-    },
-    {} as CollectionMap<CollectionWithNFT>
-  );
+        return acc;
+      }, {});
+};
 
-  return Object.values(collectionMap);
+export const getNftKey = (
+  contract: string,
+  tokenId: string,
+  currencyId: string
+): string => {
+  return `${currencyId}-${contract}-${tokenId}`;
 };
 
-export const getNftKey = (contract: string, tokenId: string): string => {
-  return `${contract}-${tokenId}`;
+const batchersMap = new Map();
+const makeBatcher = (api: API, chainId: number) =>
```

Comment:
Same here, could use a one-liner to explain what the function does (esp. since it doesn't have an explicit return type).
(But the content itself is very well commented ! :+1: )

Reply:
Done 👌